PacketDotNet/IPv6Packet.cs

```diff
@@ -184,11 +186,9 @@ public override IPProtocolType NextHeader
         /// <value>
         /// The protocol of the packet encapsulated in this ip packet
         /// </value>
-        public override IPProtocolType Protocol
-        {
-            get => NextHeader;
-            set => NextHeader = value;
-        }
+        /// 
+
+        public override IPProtocolType Protocol { get; set; }
```

Comment:
This would mean that Protocol is now modifiable without having any effect to the underlying data. The code here can be modified to look up (and set) the protocol of the actual encapsulated packet, similar to the other properties.